gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/SnappyActivation.java

```diff
@@ -198,7 +208,7 @@ protected void executeWithResultSet(SnappySelectResultSet rs)
     GfxdResultCollector<Object> rc = null;
     rc = getResultCollector(enableStreaming, rs);
     executeOnLeadNode(rs, rc, this.sql, enableStreaming, this.getConnectionID(), this.lcc
-        .getCurrentSchemaName(), this.pvs, this.isPrepStmt);
+        .getCurrentSchemaName(), this.pvs, this.isPrepStmt, this.connProps);
```

Comment:
Set "password" on GenericLanguageConnectionContext via TransactionResourceImpl like "userName" is already there and add getters for the two. Read from lcc instead of passing connProps all around.